docs/user-guide/data-structures/creating-arrays-and-datasets.ipynb

```diff
@@ -0,0 +1,527 @@
+   "source": [
+    "Or with integers which encode the number of time units elapsed since the Unix epoc.\n",
+    "See also [scipp.epoch](../../generated/functions/scipp.epoch.rst)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.datetimes(dims=['t'], values=[0, 1610288175], unit='s')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the unit is mandatory in the second case.\n",
+    "\n",
+    "`scipp.vectors` creates an array of 3-vectors.\n",
+    "It does so by converting a sequence or array with a length of 3 in its inner dimension:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.vectors(dims=['x'], values=[[1, 2, 3], [4, 5, 6]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data Arrays\n",
+    "\n",
+    "There is essentially only one way to construct [data arrays](../../generated/classes/scipp.DataArray.rst), namely its initializer:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "x = sc.linspace('x', start=1.5, stop=3.0, num=4, unit='m')\n",
+    "a = sc.scalar('an attribute')\n",
+    "m = sc.array(dims=['x'], values=[True, False, True, False])\n",
+    "data = x ** 2\n",
+    "sc.DataArray(data, coords={'x': x}, attrs={'a': a}, masks={'m': m})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`coords`, `attrs`, and `masks` are optional but the `data` must always be given.\n",
+    "Note how the creation functions for `scipp.Variable` can be used to make the individual pieces of a data array.\n",
+    "\n",
+    "## Dataset\n",
+    "\n",
+    "[Datasets](../../generated/classes/scipp.Dataset.rst) are constructed by combining multiple data arrays or variables.\n",
+    "For instance, using the previously defined variables:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.Dataset({'data1': data, 'data2': -data}, coords={'x': x})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Or from data arrays:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "da1 = sc.DataArray(data, coords={'x': x}, attrs={'a': a}, masks={'m': m})\n",
+    "da2 = sc.DataArray(-data, coords={'x': x})\n",
+    "sc.Dataset({'data1': da1, 'data2': da2})"
+   ]
+  },
```

Comment:
Add a section on the `*_like` functions (which work for variables and data arrays)?